### PR TITLE
Bug 1605352 - Add Performance FAQ to reference to regression template

### PR DESCRIPTION
**Pull request by alexandru-io**
Update regression templates to include context, and FAQs.

### CODE REVIEW

File: treeherder/perf/fixtures/performance_bug_templates.yaml [hunk 1]
```diff
@@ -6,25 +6,11 @@
...
-
-      To learn more about the regressing test(s), please see: https://wiki.mozilla.org/TestEngineering/Performance/Talos
-
-      For information on reproducing and debugging the regression, either on try or locally, see: https://wiki.mozilla.org/TestEngineering/Performance/Talos/Running
-
-      *** Please let us know your plans within 3 business days, or the offending patch(es) will be backed out! ***
-
-      Our wiki page outlines the common responses and expectations: https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling
+      Details of the alert can be found in the [alert summary]({{ alertHref }}), including links to graph and comparisons for each of the affected tests. Please follow our [guide to handling regression bugs](https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling) and **let us know your plans within 3 business days, or the offending patch(es) will be backed out** in accordance with our [regression policy](https://www.mozilla.org/en-US/about/governance/policies/regressions/).
```
Comment (davehunt) [inline treeherder/perf/fixtures/performance_bug_templates.yaml]:
```suggestion
      Details of the alert can be found in the [alert summary]({{ alertHref }}), including links to graphs and comparisons for each of the affected tests. Please follow our [guide to handling regression bugs](https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling) and **let us know your plans within 3 business days, or the offending patch(es) will be backed out** in accordance with our [regression policy](https://www.mozilla.org/en-US/about/governance/policies/regressions/).
```

File: treeherder/perf/fixtures/performance_bug_templates.yaml [hunk 2]
```diff
@@ -34,21 +20,11 @@
...
       {{ alertSummary }}
 
-      You can find links to graphs and comparison views for each of the above tests at: {{ alertHref }}
-
-      On the page above you can see an alert for each affected platform as well as a link to a graph showing the history of scores for this test. There is also a link to a treeherder page showing the jobs in a pushlog format.
-
-      To learn more about the regressing test(s), please see: https://developer.mozilla.org/en-US/docs/Mozilla/Performance/Automated_Performance_Testing_and_Sheriffing/Build_Metrics
-
-      *** Please let us know your plans within 3 business days, or the offending patch(es) will be backed out! ***
+      Details of the alert can be found in the [alert summary]({{ alertHref }}), including links to graph and comparisons for each of the affected tests. Please follow our [guide to handling regression bugs](https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling) and **let us know your plans within 3 business days, or the offending patch(es) will be backed out** in accordance with our [regression policy](https://www.mozilla.org/en-US/about/governance/policies/regressions/).
```
Comment (davehunt) [inline treeherder/perf/fixtures/performance_bug_templates.yaml]:
```suggestion
      Details of the alert can be found in the [alert summary]({{ alertHref }}), including links to graphs and comparisons for each of the affected tests. Please follow our [guide to handling regression bugs](https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling) and **let us know your plans within 3 business days, or the offending patch(es) will be backed out** in accordance with our [regression policy](https://www.mozilla.org/en-US/about/governance/policies/regressions/).
```

File: treeherder/perf/fixtures/performance_bug_templates.yaml [hunk 3]
```diff
@@ -58,19 +34,11 @@
...
+      Perfherder has detected a performance regression from push {{ revisionHref }}. As author of one of the patches included in that push, we need your help to address this regression.
 
       {{ alertSummary }}
 
-      You can find links to graphs and comparison views for each of the above tests at: {{ alertHref }}
-
-      On the page above you can see an alert for each affected platform as well as a link to a graph showing the history of scores for this test. There is also a link to a treeherder page showing the jobs in a pushlog format.
-
-      To learn more about the regressing test(s), please see: https://wiki.mozilla.org/AWSY/Tests
+      Details of the alert can be found in the [alert summary]({{ alertHref }}), including links to graph and comparisons for each of the affected tests. Please follow our [guide to handling regression bugs](https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling) and **let us know your plans within 3 business days, or the offending patch(es) will be backed out** in accordance with our [regression policy](https://www.mozilla.org/en-US/about/governance/policies/regressions/).
```
Comment (davehunt) [inline treeherder/perf/fixtures/performance_bug_templates.yaml]:
```suggestion
      Details of the alert can be found in the [alert summary]({{ alertHref }}), including links to graphs and comparisons for each of the affected tests. Please follow our [guide to handling regression bugs](https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling) and **let us know your plans within 3 business days, or the offending patch(es) will be backed out** in accordance with our [regression policy](https://www.mozilla.org/en-US/about/governance/policies/regressions/).
```

File: treeherder/perf/fixtures/performance_bug_templates.yaml [hunk 4]
```diff
@@ -79,19 +47,11 @@
...
 
       {{ alertSummary }}
 
-      You can find links to graphs and comparison views for each of the above tests at: {{ alertHref }}
-
-      On the page above you can see an alert for each affected platform as well as a link to a graph showing the history of scores for this test. There is also a link to a treeherder page showing the jobs in a pushlog format.
-
-      To learn more about the regressing test(s), please see: https://developer.mozilla.org/en-US/docs/Mozilla/Performance/Automated_Performance_Testing_and_Sheriffing/Platform_Microbenchmarks
+      Details of the alert can be found in the [alert summary]({{ alertHref }}), including links to graph and comparisons for each of the affected tests. Please follow our [guide to handling regression bugs](https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling) and **let us know your plans within 3 business days, or the offending patch(es) will be backed out** in accordance with our [regression policy](https://www.mozilla.org/en-US/about/governance/policies/regressions/).
```
Comment (davehunt) [inline treeherder/perf/fixtures/performance_bug_templates.yaml]:
```suggestion
      Details of the alert can be found in the [alert summary]({{ alertHref }}), including links to graphs and comparisons for each of the affected tests. Please follow our [guide to handling regression bugs](https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling) and **let us know your plans within 3 business days, or the offending patch(es) will be backed out** in accordance with our [regression policy](https://www.mozilla.org/en-US/about/governance/policies/regressions/).
```

File: treeherder/perf/fixtures/performance_bug_templates.yaml [hunk 5]
```diff
@@ -100,20 +60,8 @@
...
-
-      On the page above you can see an alert for each affected platform as well as a link to a graph showing the history of scores for this test. There is also a link to a Treeherder page showing the Raptor jobs in a pushlog format.
-
-      To learn more about the regressing test(s) or reproducing them, please see: https://wiki.mozilla.org/TestEngineering/Performance/Raptor
-
-      *** Please let us know your plans within 3 business days, or the offending patch(es) will be backed out! ***
-
-      Our wiki page outlines the common responses and expectations: https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling
+      Details of the alert can be found in the [alert summary]({{ alertHref }}), including links to graph and comparisons for each of the affected tests. Please follow our [guide to handling regression bugs](https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling) and **let us know your plans within 3 business days, or the offending patch(es) will be backed out** in accordance with our [regression policy](https://www.mozilla.org/en-US/about/governance/policies/regressions/).
```
Comment (davehunt) [inline treeherder/perf/fixtures/performance_bug_templates.yaml]:
```suggestion
      Details of the alert can be found in the [alert summary]({{ alertHref }}), including links to graphs and comparisons for each of the affected tests. Please follow our [guide to handling regression bugs](https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling) and **let us know your plans within 3 business days, or the offending patch(es) will be backed out** in accordance with our [regression policy](https://www.mozilla.org/en-US/about/governance/policies/regressions/).
```

File: ui/perfherder/alerts/StatusDropdown.jsx [hunk 6]
```diff
@@ -63,6 +63,7 @@ export default class StatusDropdown extends React.Component {
     }
 
     const templateArgs = {
+      revisionId: alertSummary.revision,
```
Comment (davehunt) [inline ui/perfherder/alerts/StatusDropdown.jsx]: Where is this used?

**Comment by alexandru-io** [comment]
> I don't see the reference to the FAQ. Also, as the text of all of these is now the same is there a way to make this the default and reduce the duplication?

Please see [this comment](https://github.com/mozilla/treeherder/pull/5891#discussion_r395795266)

**Comment by ionutgoldan** [comment]
@alexandru-io could you paste a markdown sample for a particular alert summary?

File: ui/perfherder/alerts/StatusDropdown.jsx (hunk 6 above)
Comment (alexandru-io) [reply ui/perfherder/alerts/StatusDropdown.jsx]: First sentence of the description, I just updated the PR.

**Comment by alexandru-io** [comment]
> @alexandru-io could you provide a markdown example for a random alert summary?

```
Perfherder has detected a performance regression from push [bb19bd67310fd0f0b309759d530ce6292b42d16e](https://hg.mozilla.org/integration/autoland/pushloghtml?changeset=bb19bd67310fd0f0b309759d530ce6292b42d16e). As author of one of the patches included in that push, we need your help to address this regression.

Regressions:

   2%  startup_about_home_paint_realworld_webextensions linux64-shippable-qr opt e10s stylo     1,137.12 -> 1,161.92


Details of the alert can be found in the [alert summary](http://localhost:5000/perf.html#/alerts?id=25465), including links to graphs and comparisons for each of the affected tests. Please follow our [guide to handling regression bugs](https://wiki.mozilla.org/TestEngineering/Performance/Talos/RegressionBugsHandling) and **let us know your plans within 3 business days, or the offending patch(es) will be backed out** in accordance with our [regression policy](https://www.mozilla.org/en-US/about/governance/policies/regressions/).

For more information on performance sheriffing please see our [FAQ](https://wiki.mozilla.org/TestEngineering/Performance/FAQ).
```

Screenshot from 2020-03-24 15-14-46

**Comment by alexandru-io** [comment]
I need to update guide to handling regression bugs as this is only for talos.

**Comment by alexandru-io** [comment]
@sarah-clements don't merge this yet, please.

**Comment by sarah-clements** [comment]
> @sarah-clements don't merge this yet, please.

I wasn't planning to. @ionutgoldan  usually merges pr's for your team.

File: treeherder/perf/fixtures/performance_bug_templates.yaml [hunk 7]
```diff
@@ -6,25 +6,13 @@
     default_component: Performance
     default_product: Testing
     text: |
-      Talos has detected a Firefox performance regression from push:
-
-      {{ revisionHref }}
-
-      As author of one of the patches included in that push, we need your help to address this regression.
+      Perfherder has detected a performance regression from push [{{ revisionId }}]({{ revisionHref }}). As author of one of the patches included in that push, we need your help to address this regression.
```
Comment (ionutgoldan) [inline treeherder/perf/fixtures/performance_bug_templates.yaml:9]: We're not mentioning which perf framework this originated on. How will engineers know where to navigate from the https://wiki.mozilla.org/TestEngineering/Performance/Handling_regression_bugs link?

File: treeherder/perf/fixtures/performance_bug_templates.yaml [hunk 8]
```diff
@@ -6,25 +6,13 @@
...
 
       {{ alertSummary }}
 
-      You can find links to graphs and comparison views for each of the above tests at: {{ alertHref }}
-
-      On the page above you can see an alert for each affected platform as well as a link to a graph showing the history of scores for this test. There is also a link to a treeherder page showing the Talos jobs in a pushlog format.
-
-      To learn more about the regressing test(s), please see: https://wiki.mozilla.org/TestEngineering/Performance/Talos
```
Comment (ionutgoldan) [inline treeherder/perf/fixtures/performance_bug_templates.yaml:21]: This Talos description is pretty valuable. It's not mentioned anywhere in https://wiki.mozilla.org/TestEngineering/Performance/Handling_regression_bugs

We should update that page to include https://wiki.mozilla.org/TestEngineering/Performance/Talos also.

File: ui/perfherder/alerts/StatusDropdown.jsx (hunk 6 above)
Comment (ionutgoldan) [reply ui/perfherder/alerts/StatusDropdown.jsx:66]: Why not simply `revision`? `revisionId` could falsely hint this as a database/implementation detail...

**Comment by ionutgoldan** [comment]
We also need to stabilize all Travis tests.

File: treeherder/perf/fixtures/performance_bug_templates.yaml (hunk 8 above)
Comment (alexandru-io) [reply treeherder/perf/fixtures/performance_bug_templates.yaml:21]: I think @davehunt can tell you more about this.

Comment (alexandru-io) [reply treeherder/perf/fixtures/performance_bug_templates.yaml:21]: @ionutgoldan not sure what can we do here as https://wiki.mozilla.org/TestEngineering/Performance/Talos is outdated and very long. I wouldn't block this PR until that page gets up to date, could take months. What's your proposal?

Comment (ionutgoldan) [reply treeherder/perf/fixtures/performance_bug_templates.yaml:21]: Sorry, I wanted to share the https://wiki.mozilla.org/TestEngineering/Performance/Talos/Tests link. I know this is the 1st thing I share to Firefox engineers for Talos-detected regressions.

Comment (alexandru-io) [reply treeherder/perf/fixtures/performance_bug_templates.yaml:21]: We can update [Talos regressions](https://wiki.mozilla.org/TestEngineering/Performance/Handling_regression_bugs#Talos_regressions) to contain this link. The idea is to have a single wiki page to reference the documentation pages of all frameworks.

Comment (alexandru-io) [reply treeherder/perf/fixtures/performance_bug_templates.yaml:21]: @ionutgoldan good enough? https://wiki.mozilla.org/TestEngineering/Performance/Handling_regression_bugs#Talos_regressions

**Comment by alexandru-io** [comment]
@ionutgoldan for some reason the Travis tests stabilized. Please review.

**Pull request merged**